k4Reco/Overlay/components/OverlayTiming.h

```diff
@@ -0,0 +1,157 @@
+#include "Gaudi/Parsers/Factory.h"
+#include "Gaudi/Property.h"
+
+#include <random>
+
+struct EventHolder {
+  std::vector<std::vector<std::string>> m_fileNames;
+  std::vector<podio::Reader>            m_rootFileReaders;
+  std::vector<int>                      m_totalNumberOfEvents;
+  std::map<int, podio::Frame>           m_events;
+
+  int m_nextEntry = 0;
+
+  EventHolder(const std::vector<std::vector<std::string>>& fileNames) : m_fileNames(fileNames) {
+    for (auto& names : m_fileNames) {
+      m_rootFileReaders.emplace_back(podio::makeReader(names));
+      m_totalNumberOfEvents.push_back(m_rootFileReaders.back().getEntries("events"));
+    }
+  }
+  EventHolder() = default;
+
+  // TODO: Cache functionality
+  // podio::Frame& read
+
+  size_t size() const { return m_fileNames.size(); }
+};
+
+using retType =
+    std::tuple<edm4hep::MCParticleCollection, std::vector<edm4hep::SimTrackerHitCollection>,
+               std::vector<edm4hep::SimCalorimeterHitCollection>, std::vector<edm4hep::CaloHitContributionCollection>>;
+
+struct OverlayTiming : public k4FWCore::MultiTransformer<retType(
+                           const edm4hep::EventHeaderCollection& headers, const edm4hep::MCParticleCollection&,
+                           const std::vector<const edm4hep::SimTrackerHitCollection*>&,
+                           const std::vector<const edm4hep::SimCalorimeterHitCollection*>&
+                           // const std::map<std::string, const edm4hep::CaloHitContributionCollection&>&
+                           )> {
+  OverlayTiming(const std::string& name, ISvcLocator* svcLoc)
+      : MultiTransformer(
+            name, svcLoc,
+            {KeyValues("EventHeader", {"EventHeader"}), KeyValues("MCParticles", {"DefaultMCParticles"}),
+             KeyValues("SimTrackerHits", {"DefaultSimTrackerHits"}),
+             KeyValues("SimCalorimeterHits", {"DefaultSimCalorimeterHits"})},
+            {KeyValues("OutputMCParticles", {"NewMCParticles"}), KeyValues("OutputSimTrackerHits", {"MCParticles1"}),
+             KeyValues("OutputSimCalorimeterHits", {"MCParticles2"}),
+             KeyValues("OutputCaloHitContributions", {"OverlayCaloHitContributions"})}) {}
+
+  template <typename T> void overlayCollection(std::string collName, const podio::CollectionBase& inColl);
+
+  virtual StatusCode initialize() final;
+
+  retType virtual operator()(
+      const edm4hep::EventHeaderCollection& headers, const edm4hep::MCParticleCollection& mcParticles,
+      const std::vector<const edm4hep::SimTrackerHitCollection*>&     simTrackerHits,
+      const std::vector<const edm4hep::SimCalorimeterHitCollection*>& simCalorimeterHits) const final;
+
+  std::pair<float, float> define_time_windows(const std::string& Collection_name) const;
+
+private:
+  // These correspond to the index position in the argument list
+  constexpr static int TRACKERHIT_INDEX_POSITION = 2;
+  constexpr static int SIMCALOHIT_INDEX_POSITION = 3;
```

Comment:
```suggestion
  constexpr static int CALOHIT_INDEX_POSITION = 3;
```
Or add SIM to the other one

Reply:
Added SIM to the other one